rest_api.py

```diff
@@ -1,11 +1,13 @@
 """REST API."""
-
 import json
-from file_adapter import FileAdapter
-from filter import filter_releases
-from filter import sort
-from flask_restful import reqparse
-from flask_restful import Resource
+import sys
+sys.path.append('../')
+from file_adapter import FileAdapter   # pylint: disable=g-import-not-at-top
+from filter_releases import filter_releases   # pylint: disable=g-import-not-at-top
+from filter_releases import sort   # pylint: disable=g-import-not-at-top
+from flask_restful import reqparse   # pylint: disable=g-import-not-at-top
+from flask_restful import Resource   # pylint: disable=g-import-not-at-top
+from airflow_connector import query_airflow  # pylint: disable=g-bad-import-order
 
 adapter = FileAdapter('fake_release_data.json', 'fake_task_data.json')
```

Comment:
A better abstraction is'

// 1. Builder paradigm
adapter = FileAdapter().loadRelease('fake_release_data.json').loadTasks('fake_task_data.json')

// 2. Old fashopm
adapter = FileAdapter()
adapter.loadRelease('fake_release_data.json')
adapter.loadTasks('fake_task_data.json')

However, why two files? Shouldn't the tasks own by the releases?

Reply:
With regards to having two files, I seem to remember that either you or Seb asked us to split them into two different sources, with the releases containing only references to the tasks which the release contained, in order to lessen the amount of data which we send to the release-ui homepage, as we don't need to have all of the task data present initially, especially considering the tasks is where a large portion of the data for a release resides

Reply:
They are different things actually....
1. Release dashboard shows a list of releases with some basic information like current state and last failed task, if any.
2. Release detail page shows all details about the release, including tasks.

Don't we already need info from #2 for #1 above? Separating them is mainly a UI concern, as we cannot dump too much info on screen at one time.

Assuming the release filter is sensible (i.e. last 7 days), sending all release + tasks from server to UI is not a lot of data. And if the result set is huge, pagination should kicks in.

Reply:
I believe that Emilia's PR changes the data structure significantly, making the concern of the particular data files somewhat irrelevant - however I believe that the same idea regarding holding only the task IDs in a release object as opposed to all of the task objects inside of release object holds true.

Additionally, the data which we will be receiving from the SQL database will be very similar to this to begin with, as the releases and task instances are in different tables. Obviously the adapter is there specifically to adapt the raw data from SQL to our own data structure.

Ultimately, it seems as if we have switched between how we store releases and tasks a few times, as you and @sebastienvas seem to have differing opinions on how release data should be represented. It would be great if we could agree on one before continuing.